Merge TxBlaster into docker image

blackjok3rtt opened the pull request:
I had to disable all the volumes. I don't know how they work. Also I guess this image needs to be made properly as you cannot build the image locally on cloud instances. 

Currently the `amount` `address` and `passphrase` are all just put into a file and copied into the image when you run `docker build`, maybe these need to be changed to ENV vars on the `ac` docker container. Really isn't much work to do this if you feel that's a better way to go, but they are fixed variables for all chains either way. 

I did some reading of docker documentation and its fine to put a couple of process inside one container as the TxBlaster is an application that uses 2 executables and a few scripts that link them together.
The mining node is also using the TxBlaster image, you just dont pass it `-blocknotify` on launch and it all it will do is mine.

Commit:
modified:   Dockerfile
	modified:   confd/templates/komodo.conf.tmpl

Commit:
Merge pull request #1 from patchkez/master

mining working

Commit:
modified:   docker-compose.yml
	modified:   komodod/Dockerfile

Commit:
Merge branch 'master' of https://github.com/blackjok2rtt/scaletest_containers

Commit:
Change miner back to seed. Place blocknotify etc

remove marketmaker, as this is now inside the container with komodod. 
Change the miner back to the seed node, this is less likley to crash, the mining bug is well and truly fixed now.
Change TxBlaster to not restart on fail, as this won't to anything. Script triggers on block heights, if it fails it fails.

Commit:
modified:   docker-compose.yml
	modified:   komodod/Dockerfile
	new file:   komodod/TxBlast
	new file:   komodod/amount
	new file:   komodod/blocknotify.sh
	new file:   komodod/getblockcount
	new file:   komodod/importprivkey
	new file:   komodod/marketmaker
	new file:   komodod/passphrase
	new file:   komodod/sendfunds
	new file:   komodod/startMM

Commit:
Update instructions

Commit:
formatting

Commit:
Update README.md

patchkez commented on README.md:
````diff
@@ -8,9 +8,9 @@
-2 images will be built, komodod and marketmaker:
+Build the TxBlaster Image:
 ```
-docker-compose build marketmaker
````
Please leave building of images via docker-compose there. docker-compose.yml knows how to build/name the image.

patchkez commented on README.md:
````diff
@@ -22,3 +22,12 @@ docker network create komodo_scale
 docker-compose up -d
 ```
+
+The seed node will launch and listen for connections on its P2P_PORT, once the ac image has connected to the miner image, the miner will start mining blocks.
````
If possible, try to follow proper terminology. Image is built from Dockerfile and Container is started from image.
_once the ac image has connected to the miner image_ - change image->container

patchkez commented on docker-compose.yml:
```diff
@@ -8,22 +8,22 @@ services:
         KOMODO_BRANCH: dev
     # image: dragonriders/kmdplatform_komodod_dev:latest
-    image: kmdplatform_komodod_dev
-    restart: on-failure
```
Why did you remove this line? We definitely need to autorestart container if process crashed.

blackjok3rtt replied:
The way the blaster works, if it fails restarting it will do nothing. It will not blast. There is no use having this here. Its better to have it totally fail, and we can then investigate why. If this is a common occurrence in small scale testing, I will need to look at ways of changing how the blast loops are called. I have a few ideas, but if this very simple way works 90%+ of the time I would rather not change it.

patchkez commented on docker-compose.yml:
```diff
@@ -8,22 +8,22 @@ services:
     ports:
     #  - "32200:32200"
       - "32199:32199"
-    volumes:
```
Please add all volumes back. What you need to do is to modify .env file and point to some directory on your disk.

patchkez commented on docker-compose.yml:
```diff
@@ -8,22 +8,22 @@ services:
       - ASSET_RPC_PASSWORD=RPcp4ZZw0rD
       - ASSET_RPC_WORKQUEUE=64
-        # - ASSET_BIND=127.0.0.1
-      - ASSET_BIND=0.0.0.0
```
Binding to all addresses was correct. Miner and AC containers will be running on their own private IP addresses. Seed node must listen to 0.0.0.0 so Miner can make connections to it.

blackjok3rtt replied:
No the miner is the one who listens. :) This is right.

patchkez commented on docker-compose.yml:
```diff
@@ -62,40 +64,19 @@ services:
       -ac_supply=77000000
       -ac_cc=1
       -ac_reward=100000000
```
End quote is missing?

patchkez commented on komodod/amount:
```diff
@@ -0,0 +1 @@
+amount=2000
```
May I ask you to completely remove this file and pass it as ENV variable? In case we need to change it, we would need to rebuild image.

patchkez commented on komodod/confd/templates/komodo.conf.tmpl:
```diff
@@ -2,6 +2,7 @@ rpcuser={{ or (getenv "KOMODO_RPC_USER") "rpcuser" }}
 rpcpassword={{ or (getenv "KOMODO_RPC_PASSWORD") "rpcpassword" }}
 server=1
 txindex=1
+blocknotify=/home/komodo/ScaleTestV2/blocknotify.sh %s
```
If we keep this line here, both AC and Miner containers will execute this script. If this is not result what we want, we need to pass it as `-blocknotify=...` from docker-compose.yml file.

blackjok3rtt replied:
This actually does nothing, it's only used as komodo.conf and didnt actually work for assets. I thought I already removed it lol.

patchkez commented on komodod/getblockcount:
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+chain=$ASSET_NAME
+komodo-cli -ac_name=$chain getblockcount
```
Can we remove all small files and make it as functions inside txblaster script?

patchkez commented on komodod/importprivkey:
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+chain=$ASSET_NAME
+komodo-cli -ac_name=$chain importprivkey Ut1Xx8tAePPxJgTkEVAwj3ZhRvBZ7gY5gJ5ob4QAPiCDcyMJEYgH
```
Please pass privkey as ENV variable. Again, we do not want to rebuild images in case this would change.

patchkez commented on komodod/passphrase:
```diff
@@ -0,0 +1 @@
+passphrase=testpassphraseforsmk702test
```
Please remove this file and pass it again as ENV.

patchkez commented on komodod/sendfunds:
```diff
@@ -0,0 +1,6 @@
+#send transaction and return TXID
+chain=$ASSET_NAME
+source amount
+komodo-cli -ac_name=$chain sendtoaddress RSNWwEWTTFH13LzWVp4EoPscKNxsT5bazt $amount
```
You know what to do here :-)

patchkez commented on komodod/sendfunds:
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Can this be function of txblaster script?

Commit:
modified:   README.md
        modified:   docker-compose.yml
	modified:   komodod/TxBlast
	modified:   komodod/blocknotify.sh
	modified:   komodod/confd/templates/komodo.conf.tmpl
	deleted:    marketmaker/Dockerfile
	deleted:    marketmaker/bin/init

Commit:
modified:   docker-compose.yml
	modified:   komodod/Dockerfile

Commit:
deleted:    amount
	deleted:    getblockcount
	deleted:    importprivkey
	deleted:    passphrase
	deleted:    sendfunds
	deleted:    startMM

Commit:
modified:   .env
	modified:   docker-compose.yml
	modified:   komodod/blocknotify.sh

Commit:
modified:   docker-compose.yml

Commit:
modified:   komodod/Dockerfile

Commit:
modified:   docker-compose.yml

Commit:
Update README.md

Commit:
modified:   docker-compose.yml
	modified:   komodod/blocknotify.sh

Commit:
Merge branch 'master' of https://github.com/blackjok3rtt/scaletest_containers

Commit:
modified:   komodod/blocknotify.sh

Commit:
modified:   komodod/blocknotify.sh

Commit:
modified:   komodod/blocknotify.sh

Commit:
modified:   komodod/blocknotify.sh

Commit:
modified:   komodod/Dockerfile

Commit:
modified:   komodod/Dockerfile

Commit:
modified:   docker-compose.yml

Commit:
modified:   docker-compose.yml

Commit:
modified:   docker-compose.yml
	modified:   komodod/TxBlast

Commit:
modified:   komodod/TxBlast
